presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -406,6 +407,8 @@ private SubqueryExpression uncoercedSubquery(SubqueryExpression subquery)
     private List<Expression> coercionsFor(Expression expression)
     {
         return analysis.getCoercions().keySet().stream()
+                .map(NodeRef::getNode)
+                // TODO why is this comparing by equality?
```

Comment:
we are gathering all coercions for a given subquery: e.g:
```
SELECT DECIMAL 1.0 * (SELECT 1 FROM ...), FLOAT 1.0 * (SELECT 1 FROM)
```
There would be two nodes for: `(SELECT 1 FROM)`, but we can gather all coercions for expression of `(SELECT 1 FROM)` in order to execute subquery only once, but with two coercion results.